managed/CounterStrikeSharp.API/Core/Capabilities/PlayerCapability.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections.Generic;
+
+namespace CounterStrikeSharp.API.Core.Capabilities;
+
+public sealed class PlayerCapability<T> where T : class
+{
+    public string Name { get; }
+    internal static readonly Dictionary<string, List<Func<CEntityInstance, T?>>> Providers = new();

```

Comment:
why CEntityInstance and not CBasePlayerController?

Reply:
Allows the implementation to further be extended later down the line to accept other entity types, so eventually you could have `RegisterPawnCapability` for instance, or any generic `RegisterCapability<T>` where `T : CEntityInstance`. In that way plugin authors could attach any extra data they want to a `CPlantedC4` for instance.

This is all hidden away from the end user at this point as we are supporting purely player & plugin at this stage.

Reply:
I suppose in this case it is a sealed `PlayerCapability` class so it should probably be changed to `CCSPlayerController`